src/iptvsimple/data/Channel.cpp

```diff
@@ -60,10 +64,32 @@ void Channel::Reset()
   m_encryptionSystem = 0;
   m_tvgShift = 0;
   m_channelName.clear();
-  m_logoPath.clear();
+  m_iconPath.clear();
   m_streamURL.clear();
   m_tvgId.clear();
   m_tvgName.clear();
-  m_tvgLogo.clear();
   m_properties.clear();
-}
\ No newline at end of file
+}
+
+void Channel::SetIconPathFromTvgLogo(const std::string& tvgLogo, std::string& channelName)
+{
+  m_iconPath = tvgLogo;
+
+  bool logoSetFromChannelName = false;
+  if (m_iconPath.empty())
+  {
+    m_iconPath = m_channelName;
+    logoSetFromChannelName = true;
+  }
+
+  m_iconPath = XBMC->UnknownToUTF8(m_iconPath.c_str());
+
+  // urlencode channel logo when set from channel name and source is Remote Path
+  // append extension as channel name wouldn't have it
+  if (Settings::GetInstance().GetLogoPathType() == PathType::REMOTE_PATH && logoSetFromChannelName)
+    m_iconPath = utilities::WebUtils::UrlEncode(m_iconPath) + CHANNEL_LOGO_EXTENSION;
+
+  const std::string& logoLocation = Settings::GetInstance().GetLogoLocation();
+  if (!logoLocation.empty() && m_iconPath.find("://") == std::string::npos) // not absolute path
+    m_iconPath = utilities::FileUtils::PathCombine(logoLocation, m_iconPath);
```

Comment:
You don't need `logoLocation ` if `m_iconPath.find("://") == std::string::npos` evaluates to `false`.

```c++
  if (m_iconPath.find("://") == std::string::npos)
  {
    const std::string& logoLocation = Settings::GetInstance().GetLogoLocation();
    if (!logoLocation.empty())
    {
      // not absolute path
      m_iconPath = utilities::FileUtils::PathCombine(logoLocation, m_iconPath);
    }
  }
```